code/game/objects/items/tools/spess_knife.dm

```diff
@@ -0,0 +1,144 @@
+/obj/item/spess_knife
+	name = "spess knife"
+	desc = "An all-in-one tool combining knife, screwdriver, cutters and many more."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "spess_knife"
+	inhand_icon_state = "screwdriver" // TODO
+	worn_icon_state = "screwdriver" // TODO
+	belt_icon_state = "screwdriver" // TODO
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	flags_1 = CONDUCT_1
+	slot_flags = ITEM_SLOT_BELT
+	resistance_flags = FIRE_PROOF
+	tool_behaviour = null
+	toolspeed = 2 // Two times worse than default tools
+	custom_materials = list(/datum/material/iron=SMALL_MATERIAL_AMOUNT)
+	hitsound = SFX_SWING_HIT
+	///Chance to select wrong tool
+	var/wrong_tool_prob = 10
+
+/obj/item/spess_knife/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/butchering, \
+		speed = 12 SECONDS, \
+		effectiveness = 80, \
+		disabled = TRUE, \
+	)
+
+/obj/item/spess_knife/attack_self(mob/user, modifiers)
+	var/list/options = list(
+		"fold" = image(icon = 'icons/obj/tools.dmi', icon_state = initial(icon_state)),
+		TOOL_KNIFE = image(icon = 'icons/obj/tools.dmi', icon_state = "[initial(icon_state)]_[TOOL_KNIFE]"),
+		TOOL_SCREWDRIVER = image(icon = 'icons/obj/tools.dmi', icon_state = "[initial(icon_state)]_[TOOL_SCREWDRIVER]"),
+		TOOL_WIRECUTTER = image(icon = 'icons/obj/tools.dmi', icon_state = "[initial(icon_state)]_[TOOL_WIRECUTTER]"),
+		)
+	var/new_tool_behaviour = show_radial_menu(user, src, options, require_near = TRUE, tooltips = TRUE)
+	if(isnull(new_tool_behaviour) || new_tool_behaviour == tool_behaviour)
+		return ..()
```

Comment:
I'd suggest calling parent at the start of the proc and respecting return value with `. = ..() if(.) return`, IIRC it won't have any effect on the use but it sends signal (good) and applies fingerprints for you.